src/main/java/hudson/plugins/jira/Updater.java

```diff
@@ -127,22 +127,38 @@ static void submitComments(
             boolean useWikiStyleComments, boolean recordScmChanges, String groupVisibility, String roleVisibility) throws RemoteException {
         // copy to prevent ConcurrentModificationException
         List<JiraIssue> copy = new ArrayList<JiraIssue>(issues);
+        StringBuilder issuesWithExceptions = new StringBuilder();
         for (JiraIssue issue : copy) {
-            try {
                 logger.println(Messages.Updater_Updating(issue.id));
-                session.addComment(
+                
+                try {
+                	session.addComment(
                         issue.id,
                         createComment(build, useWikiStyleComments, jenkinsRootUrl, recordScmChanges, issue),
                         groupVisibility, roleVisibility);
-            } catch (RemotePermissionException e) {
-                // Seems like RemotePermissionException can mean 'no permission' as well as
-                // 'issue doesn't exist'.
-                // To prevent carrying forward invalid issues forever, we have to drop them
-                // even if the cause of the exception was different.
-                logger.println("Looks like " + issue.id + " is no valid JIRA issue or you don't have permission to update the issue.\n" +
-                        "Issue will not be updated.\n" + e);
+                } catch (RemotePermissionException e) {
```

Comment:
It depends. In my opinion a configuration switch is needed for the decision. The current implementation comes from contributor 'kutzi', a commit with hash '2c146187a1cdf...'. I just took over.